runtime/src/main/scala/akka/grpc/internal/ClientState.scala

```diff
@@ -97,24 +98,40 @@ final class ClientState(settings: GrpcClientSettings, channelFactory: GrpcClient
 
   private def create(): InternalChannel = {
     val internalChannel: InternalChannel = channelFactory(settings)
-    internalChannel.done.onComplete {
-      case Failure(_: ClientConnectionException | _: NoTargetException) =>
-        val old = internalChannelRef.get()
-        if (old.isDefined) {
-          val newInternalChannel = create()
-          // Only one client is alive at a time. However a close() could have happened between the get() and this set
-          if (!internalChannelRef.compareAndSet(old, Some(newInternalChannel))) {
-            // close the newly created client we've been shutdown
-            ChannelUtils.close(newInternalChannel)
-          }
+    internalChannel.managedChannel.onComplete {
+      case _: Success[_] =>
+        internalChannel.done.onComplete {
+          case Failure(_: ClientConnectionException | _: NoTargetException) =>
+            // Would be better to retry with backoff
+            // Would be good to log
+            recreate()
+          case Failure(_) =>
+            // This makes the client unusable. Perhaps we should retry with backoff here too?
+            // Would be good to log
+            close()
+          case _ =>
+          // let success through
         }
       case Failure(_) =>
-        close()
-      case _ =>
-      // let success through
+        // Would be better to retry with backoff
+        // Would be good to log
+        recreate()
```

Comment:
I guess we could use `akka.pattern` `retry` for a cheap backoff, delay there is constant but better than nothing?

Reply:
I take that back, I quick skimmed and missed how it is side effecting, so probably not possible to leverage that.

Reply:
(it would also be good to have a (configurable, perhaps resetting) limit on the number of retries here)

Reply:
I think we should use the the configured delay and max retries here as well (using `akka.pattern.Patterns#after` and passing the recreateCount as a parameter to `recreateOnFailure` and `recreate` for example).

Having a backoff and max retries only apply to initial connection seems quite confusing to me.